diff --git a/docker/Dockerfile b/docker/Dockerfile
index a8a2130..05b942e 100755
--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -1,4 +1,4 @@
-FROM busybox:stable
+FROM lejianwen/rustdesk-api:latest
 
 ARG S6_OVERLAY_VERSION=3.2.0.0
 ARG S6_ARCH=x86_64
@@ -15,11 +15,13 @@ COPY rootfs /
 ENV RELAY=relay.example.com
 ENV ENCRYPTED_ONLY=0
 
-EXPOSE 21115 21116 21116/udp 21117 21118 21119
+EXPOSE 21114 21115 21116 21116/udp 21117 21118 21119
 
 HEALTHCHECK --interval=10s --timeout=5s CMD /usr/bin/healthcheck.sh
 
-WORKDIR /data
+WORKDIR /app
+
+VOLUME /app/data
 
 VOLUME /data
 
diff --git a/docker/rootfs/etc/s6-overlay/s6-rc.d/api/dependencies b/docker/rootfs/etc/s6-overlay/s6-rc.d/api/dependencies
new file mode 100644
index 0000000..f8e8634
--- /dev/null
+++ b/docker/rootfs/etc/s6-overlay/s6-rc.d/api/dependencies
@@ -0,0 +1,2 @@
+key-secret
+hbbs
diff --git a/docker/rootfs/etc/s6-overlay/s6-rc.d/api/run b/docker/rootfs/etc/s6-overlay/s6-rc.d/api/run
new file mode 100644
index 0000000..97e51bf
--- /dev/null
+++ b/docker/rootfs/etc/s6-overlay/s6-rc.d/api/run
@@ -0,0 +1,4 @@
+#!/command/with-contenv sh
+sleep 3
+cd /app
+./apimain
diff --git a/docker/rootfs/etc/s6-overlay/s6-rc.d/api/type b/docker/rootfs/etc/s6-overlay/s6-rc.d/api/type
new file mode 100644
index 0000000..5883cff
--- /dev/null
+++ b/docker/rootfs/etc/s6-overlay/s6-rc.d/api/type
@@ -0,0 +1 @@
+longrun
diff --git a/docker/rootfs/etc/s6-overlay/s6-rc.d/user/contents.d/api b/docker/rootfs/etc/s6-overlay/s6-rc.d/user/contents.d/api
new file mode 100644
index 0000000..e69de29
